playbooks/templates/.env.j2

```diff
@@ -1,118 +1,237 @@
-# STRIPE_API_KEY=
-# STRIPE_SECRET_KEY=
-# STRIPE_WEBHOOK_SECRET=
+# Used to load skylinks blocklist data from Airtable
+AIRTABLE_API_KEY={{ webportal_common_config.airtable_api_key }}
 
+{% if webportal_server_config.portal_modules is defined and webportal_server_config.portal_modules is not none and 'a' in webportal_server_config.portal_modules %}
 ###############################################################################
 # (optional) Accounts
 
-# the public IP/domain of your server, e.g. `helsinki.siasky.net
-CR_IP=<subdomain>.siasky.net
-
-# the list of servers (with ports) which make up your CockroachDB cluster, e.g.
-# `<server-1>.siasky.net:26257,<server-2>.siasky.net:26257,<server-3>.siasky.net:26257`
-CR_CLUSTER_NODES=
-
-# absolute url to the portal dashboard e.g. https://account.siasky.net
-SKYNET_DASHBOARD_URL=
-
-# this is the MongoDB username
-SKYNET_DB_USER=
-
-# this is the MongoDB password
-SKYNET_DB_PASS=
-
-# this is the MongoDB address or container name
+# Public IP/domain of server used by CoackroachDB
+# For a single server: single-portal.com
+# For a cluster: server-1.cluster.com
+CR_IP={{ server_domain }}
+
+# List of servers (with ports) which make up your CockroachDB cluster.
+# A server can have a link to itself, so the same string can be used for all
+# servers in a cluster.
+# For a single server: single-portal.com:26257
+# For a cluster: server-1.cluster.com:26257,server-2.cluster.com:26257,server-3.cluster.com:26257
+{#
+Jinja template dev notes:
+- hostvars contain data from all Ansible hosts defined in the Ansible inventory
+- Get list of hosts in the same cluster:
+  hostvars | dictsort | selectattr('1.portal_cluster_id', 'defined') | selectattr('1.portal_cluster_id', 'equalto', (portal_cluster_id | default('')))
+- Get just domain names from the list:
+  map(attribute='1.ansible_host') | list
+- Add ports to the list items:
+  product('[:26257']) | map('join') | list
+#}
+CR_CLUSTER_NODES={{
+    hostvars | dictsort | selectattr('1.portal_cluster_id', 'defined') | selectattr('1.portal_cluster_id', 'equalto', (portal_cluster_id | default(''))) | map(attribute='1.ansible_host') | list | product([':26257']) | map('join') | list | join(',')
+    if portal_cluster_id is defined
+    else server_domain + ':26257'
+}}
+
+# Absolute url to the (cluster) portal dashboard.
+# For a single server: https://account.single-portal.com
+# For a cluster: https://account.cluster.com
+SKYNET_DASHBOARD_URL=https://account.{{
+    cluster_domain
+    if portal_cluster_id is defined
+    else server_domain
+}}
+
+# MongoDB username
+SKYNET_DB_USER=admin
+
+# MongoDB password
+SKYNET_DB_PASS={{
+    webportal_common_config.skynet_db_pass
+    if portal_cluster_id is defined
+    else webportal_server_config.skynet_db_pass
+}}
+
+# MongoDB address or container name
 SKYNET_DB_HOST=mongo
 
-# this is the MongoDB port
+# MongoDB port
 SKYNET_DB_PORT=27017
 
-# this is accounts log level
+# Accounts log level
```

Comment:
I would add a note saying that this is for Jaeger.

Reply:
It is not. This sets the log level for accounts and has nothing to do with Jaeger.